libpod/adapter/runtime_remote.go

```diff
@@ -119,6 +122,42 @@ func (r *LocalRuntime) NewImageFromLocal(name string) (*ContainerImage, error) {
 
 }
 
+// LoadFromArchiveReference creates an image from a local archive
+func (r *LocalRuntime) LoadFromArchiveReference(ctx context.Context, srcRef types.ImageReference, signaturePolicyPath string, writer io.Writer) ([]*ContainerImage, error) {
+	// TODO We need to find a way to leak certDir, creds, and the tlsverify into this function, normally this would
+	// come from cli options but we don't want want those in here either.
+	imageID, err := iopodman.PullImage().Call(r.Conn, srcRef.DockerReference().String(), "", "", signaturePolicyPath, true)
```

Comment:
I can’t see how this works at all.

- `LoadFromArchiveReference` references a _local_ archive; just referencing the same path from the remote computer does not nearly do the same thing. (BTW this applies to the non-`LoadFromArchiveReference` pull code paths as well.)
- The `srcRef.DockerReference()` use completely drops the path, i.e. the _most essential aspect_ of loading from a file, in favor of a destination value which can in principle exist but is currently only ever used for destinations, i.e. this “load” implementation drops the useful path and preserves the nonsensical reference.

Reply:
@mtrmac what is the correct way to get from the srcref to a string representation of the srcRef?

Reply:
`transports.ImageName(srcRef)` outputs the round-trip-safe string representation (to be parsed by `alltransports.ParseImageName`).

Still, that does not magically transfer the local file to the other end, nor tell you that the reference depends on a local file.

Reply:
I understand, right now we only support files that are local to the backend (not the client).  We are working on a solution for this however.

Reply:
Also, for a known transport, `srcRef.StringWithinTransport()` can be parsed by `that.Transport.ParseReference()`; that does not require the broad `alltransports` dependency.